cloud/google/machineactuator.go

```diff
@@ -600,11 +629,25 @@ func (gce *GCEClient) instanceIfExists(machine *clusterv1.Machine) (*compute.Ins
 	return instance, nil
 }
 
-func (gce *GCEClient) providerconfig(providerConfig clusterv1.ProviderConfig) (*gceconfigv1.GCEProviderConfig, error) {
-	return gce.gceProviderConfigCodec.DecodeFromProviderConfig(providerConfig)
+func (gce *GCEClient) machineproviderconfig(providerConfig clusterv1.ProviderConfig) (*gceconfigv1.GCEMachineProviderConfig, error) {
```

Comment:
(idea)
I see that the cluster provider config and machine provider config get read everywhere. Perhaps it will be more sustainable to return a wrapped error in these two helper methods that give the context as to which provider config reading hits the error instead of trying to remember to wrap the errors every time these methods are called.